### ISSUE TITLE
Fatal error with Phar, Psalm 4.2.0 and PHP 7.2

### ISSUE THREAD

**Issue by ADmad** [vimeo/psalm#4640]
Using the phar as `php pslam.phar` resulted in:

```
PHP Fatal error:  Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 7.3.0". You are running 7.2.34-8+ubuntu18.04.1+deb.sury.org+1. in phar:///path/to/project/psalm.phar/vendor/composer/platform_check.php on line 24
PHP Stack trace:
PHP   1. {main}() /path/to/project/psalm.phar:0
PHP   2. require() /path/to/project/psalm.phar:14
PHP   3. require_once() phar:///path/to/project/psalm.phar/psalm:5
PHP   4. Psalm\Internal\IncludeCollector->runAndCollect() phar:///path/to/project/psalm.phar/src/psalm.php:220
PHP   5. Psalm\{closure:phar:///path/to/project/psalm.phar/src/psalm.php:218-220}() phar:///path/to/project/psalm.phar/src/Psalm/Internal/IncludeCollector.php:31
PHP   6. Psalm\requireAutoloaders() phar:///path/to/project/psalm.phar/src/psalm.php:219
PHP   7. require_once() phar:///path/to/project/psalm.phar/src/command_functions.php:46
PHP   8. ComposerAutoloaderInit37464bc06b271c99d7b604582fbbb7a2::getLoader() phar:///path/to/project/psalm.phar/vendor/autoload.php:9
PHP   9. require() phar:///path/to/project/psalm.phar/vendor/composer/autoload_real.php:25
PHP  10. trigger_error() phar:///path/to/project/psalm.phar/vendor/composer/platform_check.php:24
```

**Comment by muglug** [comment]
I think this is because the Phar is built on 7.3

**Comment by staabm** [comment]
I guess this issue relates to composer v2 new platform checking capabilities 

https://php.watch/articles/composer-platform-check

**Comment by staabm** [comment]
You might disablr this check via a composer.json config flag

https://getcomposer.org/doc/06-config.md#platform-check

**Comment by muglug** [comment]
Thanks @staabm, I'm betting that's it

**Comment by fahadul25** [comment]
Is there any solution to this problem ?

**Comment by ADmad** [comment]
The issue is resolved in 4.4.0.
